### PR TITLE
com.github.iwalton3.jellyfin-media-player

### PR DESCRIPTION
**Pull request by iwalton3**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by iwalton3** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

**Comment by iwalton3** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

File: com.github.iwalton3.jellyfin-media-player.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
+{
+  "app-id": "com.github.iwalton3.jellyfin-media-player",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.12",
```
Comment (LeandroStanger) [inline com.github.iwalton3.jellyfin-media-player.json:4]: Update runtime to 5.15
```suggestion
  "runtime-version": "5.15",
```

**Comment by iwalton3** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

**Comment by iwalton3** [comment]
Build looks good. I can download it and everything is working properly.

**Comment by iwalton3** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

**Comment by iwalton3** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

File: com.github.iwalton3.jellyfin-media-player.json [hunk 2]
```diff
@@ -0,0 +1,275 @@
+{
+  "app-id": "com.github.iwalton3.jellyfin-media-player",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15",
+  "sdk": "org.kde.Sdk",
+  "base": "io.qt.qtwebengine.BaseApp",
+  "base-version": "5.15",
+  "command": "jellyfinmediaplayer",
+  "finish-args": [
+    "--socket=x11",
```
Comment (barthalion) [inline com.github.iwalton3.jellyfin-media-player.json:10]:
```suggestion
    "--socket=fallback-x11",
```

File: com.github.iwalton3.jellyfin-media-player.json [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    "--share=network",
+    "--device=dri",
+    "--device=all",
+    "--talk-name=org.freedesktop.PowerManagement",
+    "--talk-name=org.freedesktop.ScreenSaver",
+    "--system-talk-name=org.freedesktop.login1"
+  ],
+  "build-options": {
+    "strip": true
+  },
```
Comment (barthalion) [inline com.github.iwalton3.jellyfin-media-player.json:23]:
```suggestion
  "build-options": {
    "strip": true
  },
```

This is the default.

File: com.github.iwalton3.jellyfin-media-player.json [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+            "python3 waf build",
+            "python3 waf install"
+          ],
+          "cleanup": [ "/lib/pkgconfig", "/share", "/include" ],
+          "sources": [
+            {
+              "type": "archive",
+              "url": "https://github.com/mpv-player/mpv/archive/v0.29.1.tar.gz",
+              "sha256": "f9f9d461d1990f9728660b4ccb0e8cb5dce29ccaa6af567bec481b79291ca623"
+            },
```
Comment (barthalion) [inline com.github.iwalton3.jellyfin-media-player.json:68]: mpv seems to be defined twice, here and few lines below.

File: com.github.iwalton3.jellyfin-media-player.json [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+              "sources": [
+                {
+                  "type": "archive",
+                  "url": "https://github.com/FFmpeg/nv-codec-headers/releases/download/n8.2.15.8/nv-codec-headers-8.2.15.8.tar.gz",
+                  "sha256": "770b20b7e63adf84d42394b50425cf9ac7925913ac4f77faba552aa525dd437a"
+                }
+              ]
+            },
+            {
+              "name": "ffmpeg",
```
Comment (barthalion) [inline com.github.iwalton3.jellyfin-media-player.json:94]: Also duplicated below.

File: com.github.iwalton3.jellyfin-media-player.json [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+  "base-version": "5.15",
+  "command": "jellyfinmediaplayer",
+  "finish-args": [
+    "--socket=x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--share=ipc",
+    "--share=network",
+    "--device=dri",
+    "--device=all",
```
[the remark this answers is not in the document]
Comment (iwalton3) [reply com.github.iwalton3.jellyfin-media-player.json]: I believe it's required for hardware video decoding. I copied it from the PlexMediaPlayer Flatpak I based it off.

File: com.github.iwalton3.jellyfin-media-player.json (hunk 5 above)
Comment (iwalton3) [reply com.github.iwalton3.jellyfin-media-player.json:93]: One has Nvidia hardware acceleration and only works on x86 and the other doesn't. Is there a better way to do this?

**Comment by nedrichards** [comment]
bot, build com.github.iwalton3.jellyfin-media-player

File: com.github.iwalton3.jellyfin-media-player.json (hunk 5 above)
Comment (nedrichards) [reply com.github.iwalton3.jellyfin-media-player.json:90]: No, that makes sense to me.

**Pull request closed without merging**